deploy-argocd.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash -e
+
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+
+main() {
+    verify_permissions || exit $?
+    install_argocd
+    wait_for_pods_ready
+    port_forward_argocd
+    login_argocd
+    deploy_apps
+}
+
+verify_permissions() {
+    if [ "$(oc auth can-i '*' '*' --all-namespaces)" != "yes" ]; then
+        echo
+        echo "[ERROR] User '$(oc whoami)' does not have the required 'cluster-admin' role." 1>&2
+        echo "Log into the cluster with a user with the required privileges (e.g. kubeadmin) and retry."
+        return 1
+    fi
+}
+
+install_argocd() {
+
+    echo "Installing the Argo CD operator subscription:"
+    kubectl apply -k "$ROOT/konflux-ci/argocd"
+    echo -n "Waiting for default project (and namespace) to exist: "
+    while ! kubectl get appproject/default -n argocd &>/dev/null; do
+        echo -n .
+        sleep 1
+    done
+    kubectl config set-context --current --namespace=argocd
+    echo "OK"
+}
+
+wait_for_pods_ready() {
+
+    echo "Waiting for argo CD pods to be ready"
+    while ! kubectl wait --for=condition=ready pod -n argocd -l app.kubernetes.io/name=argocd-server &>/dev/null; do
+     echo -n .
+     sleep 1
+    done
+    echo "Pods are ready"
+
+}
+
+port_forward_argocd() {
+    echo -n "Port forwarding for Argo CD "
+    kubectl port-forward svc/argocd-server -n argocd 8084:443 &
+    echo "Argo CD can be accessed at https://localhost:8084/"
+}
+
+login_argocd() {
+    ARGOCD_PASSWORD=$(kubectl -n argocd get secret argocd-initial-admin-secret -o jsonpath="{.data.password}" | base64 -d && echo)
```

Comment:
I think this is the preferred approach:
```suggestion
    local argocd_password
    argocd_password=$(kubectl -n argocd get secret argocd-initial-admin-secret -o jsonpath="{.data.password}" | base64 -d && echo)
```